pkg/configs/instance_limits.go

```diff
@@ -0,0 +1,32 @@
+package configs
+
+import (
+	"errors"
+	"flag"
+	"strings"
+
+	"github.com/cortexproject/cortex/pkg/util/flagext"
+	"github.com/cortexproject/cortex/pkg/util/resource"
+)
+
+type InstanceLimits struct {
```

Comment:
I wonder if we can find a better name than `InstanceLimits`. `InstanceLimits` from the name can be any limits at instance level. But I guess this scope is just resource utilization based limits like CPU, Mem, Disk?

Reply:
I see what you mean. I first tried to find where to put in ingester, and decided to append to the `instance_limits` group to reuse the generic group.

If we create a new group for the resource utilization limit, it would be something like `-ingester.utilization-limits.cpu`. I initially thought I'm creating a new group that's too specific to this PR, but I'm open to change it.

Reply:
`InstanceLimits` is fine if we don't have better naming.